processing/src/main/java/org/apache/druid/query/aggregation/JavaScriptAggregatorFactory.java

```diff
@@ -58,8 +59,13 @@
   private final String fnCombine;
   private final JavaScriptConfig config;
 
-  // This variable is lazily initialized to avoid unnecessary JavaScript compilation during JSON serde
-  private JavaScriptAggregator.ScriptAggregator compiledScript;
+  /**
+   * The field is declared volatile in order to ensure safe publication of the object
+   * in {@link #compileScript(String, String, String)} without worrying about final modifiers
+   * on the fields of the created object
+   */
+  @Nullable
```

Comment:
`@MonotonicNonNull`

Reply:
Do you mean: https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/MonotonicNonNull.html This is not included at _druid-processing_ dependencies?

Reply:
Yes

Reply:
Just for this part, documentation says that: 

> Use of @MonotonicNonNull on a static field is a code smell: it may indicate poor design. You should consider whether it is possible to make the field a member field that is set in the constructor.

[Nullness Annotations](https://checkerframework.org/manual/#nullness-annotations)

Reply:
`compiledScript` is not static

Reply:
Same problem applies to interface import. Here is a related explanation: https://stackoverflow.com/questions/29097160/why-are-nonnull-annotation-on-fqn-not-allowed On demand static import solves the problem as follows, otherwise it gives that error:

> Static Member Qualifying Type may Not be Annotated

```java
  @MonotonicNonNull
  @Nullable
  private volatile ScriptAggregator compiledScript;
```

Reply:
It shouldnt have MonotonicNonNull AND Nullable. Just MonotonicNonNull